src/clp_ffi_js/ir/query_methods.cpp

```diff
@@ -0,0 +1,150 @@
+    }
+
+    [[nodiscard]] static auto handle_schema_tree_node_insertion(
+            [[maybe_unused]] bool is_auto_generated,
+            [[maybe_unused]] clp::ffi::SchemaTree::NodeLocator schema_tree_node_locator,
+            [[maybe_unused]] std::shared_ptr<clp::ffi::SchemaTree const> const& schema_tree
+    ) -> IRErrorCode {
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    [[nodiscard]] static auto handle_end_of_stream() -> IRErrorCode {
+        return IRErrorCode::IRErrorCode_Success;
+    }
+
+    [[nodiscard]] auto get_deserialized_log_event_indexes() const -> std::vector<size_t> const& {
+        return m_deserialized_log_event_indexes;
+    }
+
+private:
+    std::vector<size_t> m_deserialized_log_event_indexes;
+};
+
+namespace {
+auto trivial_new_projected_schema_tree_node_callback(
+        [[maybe_unused]] bool is_auto_generated,
+        [[maybe_unused]] clp::ffi::SchemaTree::Node::id_t node_id,
+        [[maybe_unused]] std::string_view projected_key_path
+) -> ystdlib::error_handling::Result<void> {
+    return ystdlib::error_handling::success();
+}
+}  // namespace
+
+[[nodiscard]] auto
+query_log_event_indices(clp::ReaderInterface& reader, std::string const& query_string)
+        -> std::vector<size_t> {
+    std::istringstream query_string_stream{query_string};
+    auto query_handler_result{
+            clp::ffi::ir_stream::search::QueryHandler<
+                    decltype(&trivial_new_projected_schema_tree_node_callback)>::
+                    create(&trivial_new_projected_schema_tree_node_callback,
+                           clp_s::search::kql::parse_kql_expression(query_string_stream),
+                           {},
+                           false)
+    };
+
+    if (query_handler_result.has_error()) {
+        auto const error_code{query_handler_result.error()};
+        throw ClpFfiJsException{
+                clp::ErrorCode::ErrorCode_Failure,
+                __FILENAME__,
+                __LINE__,
+                std::format(
+                        "Failed to create query handler: {} {}",
+                        error_code.category().name(),
+                        error_code.message()
+                )
+        };
+    }
+
+    auto deserializer_result{clp::ffi::ir_stream::make_deserializer(
+            reader,
+            LogEventIndexIrUnitHandler{},
+            std::move(query_handler_result.value())
+    )};
+
+    if (deserializer_result.has_error()) {
+        auto const error_code{deserializer_result.error()};
+        throw ClpFfiJsException{
+                clp::ErrorCode::ErrorCode_Failure,
+                __FILENAME__,
+                __LINE__,
+                std::format(
+                        "Failed to create deserializer: {} {}",
+                        error_code.category().name(),
+                        error_code.message()
+                )
+        };
+    }
+
+    auto& deserializer{deserializer_result.value()};
```

Comment:
(we previously didn't do a generic handler for unstructured and structured ir because they were vastly different. however, now) this piece of logic seems to be highly similar to the one in `StructuredIrStreamReader::deserialize_stream()`. any idea on how we can deduplicate so that the error handlings don't go out of sync?